src/BeadingStrategy/RedistributeBeadingStrategy.cpp

```diff
@@ -24,7 +24,8 @@ RedistributeBeadingStrategy::RedistributeBeadingStrategy(   const coord_t optima
 
 coord_t RedistributeBeadingStrategy::getOptimalThickness(coord_t bead_count) const
 {
-    return parent->getOptimalThickness(bead_count);
+    const coord_t inner_bead_count = bead_count > 2 ? bead_count - 2 : 0;
```

Comment:
@jellespijker 
This is problematic when `bead_count == 1` or `0`:

we then get `getOptimalThickness(1) == getOptimalThickness(2)` because in both cases `inner_bead_count == 0`.

This causes `getTransitionAnchorPos` to return `inf` which produces an `end_pos=-9223372036854775761` in `generateTransitionEnds`

Note there is potentially other cases for `getTransitionAnchorPos` to return `inf`, for example, `LimitedBeadingStrategygetTransitionThickness(bead_count) == LimitedBeadingStrategygetTransitionThickness(bead_count+1)` will return when `bead_count > max_bead_count`

Reply:
Ah sorry, I didn't realized that the next line from last commit fixed it.

Reply:
@Piezoid no worries. I missed it indeed but lucky for me @nallath caught it during component testing and fixed it.

Reply:
Ha, good to see that both of us came to the same conclusion on this :)